boards/arduino-leonardo/leonardo-runner.sh

```diff
@@ -1,6 +1,39 @@
 #!/usr/bin/env sh
 set -e
 
+LINUX_OS="LINUX"
+MAC_OS="MAC"
+UNKNOWN_OS="UNKNOWN"
+
+case "$(uname -s)" in
+    Linux*)     OS=$LINUX_OS;;
+    Darwin*)    OS=$MAC_OS;;
+    *)          OS=$UNKNOWN_OS
+esac
+
+if ! command -v numfmt &> /dev/null
+then
+    echo "required numfmt could not be found!"
+    echo "please install https://command-not-found.com/numfmt"
+    exit 1
```

Comment:
IMO `numfmt` missing shouldn't be a hard error.  Let's just add a warning and replace numfmt with a dummy function:

```suggestion
    echo "numfmt is needed for human-readable sizes." >&2
    echo "please install https://command-not-found.com/numfmt" >&2
    alias numfmt=true
```